.github/workflows/solutions.yml

```diff
@@ -11,12 +11,26 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
       - uses: actions/setup-node@v3
         with:
           node-version: "16"
           cache: "npm"
       - run: npm ci
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v22
+        with:
+          files: projects
+      - name: Run changed solutions in project directories
+        if: ${{ steps.changed-files.outputs.only_changed == 'true' }}
+        run: |
+          chmod +x ./scripts/changed-solutions.sh
```

Comment:
Same with the other PR, this should be checked in with the right permissions. That way people don't need to `chmod +x` manually.

...and I just noticed we checked in `chmod +x ./scripts/check-solutions.sh` later down in the file. Whoops! That shouldn't be there. But we can fix that in another PR.